src/workers/forum_automod.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime, timezone, timedelta
+import discord
+import logging
+import asyncio
+from settings import get_settings
+from util import log_info_and_embed, send_dm
+from .helper import create_automod_embed
+from discord.ext import tasks
+from discord.utils import snowflake_time
+
+settings = get_settings()
+logger = logging.getLogger(__name__)
+
+
+@tasks.loop(hours=24)
+async def autodelete_threads(self):
+    guild = self.get_guild(settings.guild_id)
+    if guild is None:
+        return
+
+    for channel_id, duration in settings.automod_inactivity.items():
+        channel = guild.get_channel(channel_id)
+        if channel is None:
+            continue
+
+        for thread in channel.threads:
+            last_post = thread.last_message_id
+
+            if thread.flags.pinned:
+                continue
+
+            # check if starter message was deleted
+            starter_message = None
+            try:
+                starter_message = await thread.fetch_message(thread.id)
+                await asyncio.sleep(0.5)
```

Comment:
i'm scared that we'll get rate limited if we're doing a request every 0.5 seconds for the overhead

Reply:
discord.py will handle the overall rate limit for us, (5 API requests per second is the limit). This sleep is here to prevent the task from blocking the other parts of the bot

Reply:
ok sounds good to me 👍